src/components/utility/FetchQuestionnaire.tsx

```diff
@@ -2,27 +2,48 @@ import { Alert, Platform, ToastAndroid } from 'react-native';
 import { getAllQuestionnaireDataByCode } from '../../data/api/Questionnaire';
 import ParticipantCode from '../../data/localStorage/ParticipantCode';
 import ACCESSIBILITY_STRINGS from '../../assets/accessibilityStrings';
+
 let isFetching = false;
 
 export async function fetchQuestionnaire(code: string, navigation: any) {
   if (!isFetching) {
     isFetching = true;
 
-    let questionnaire = await getAllQuestionnaireDataByCode(code);
+    let response = await getAllQuestionnaireDataByCode(code);
 
-    if (questionnaire) {
+    if (response && !response.error) {
       await Promise.all([
         ParticipantCode.saveCurrentParticipantCodeToLocalStorage(code),
         ParticipantCode.addQuestionnaireInLocalStorage({
           code: code,
-          name: questionnaire.title,
+          name: response.data.title,
         }),
       ]);
+      isFetching = false;
       // @ts-ignore next-line
       navigation.navigate('Questionnaire', {
-        title: questionnaire.title,
-        questionnaire: questionnaire,
+        title: response.data.title,
+        questionnaire: response.data,
       });
+      return true;
+    } else if (response && response.error) {
+      // probably an error
+      if (response.status === 404) {
+        if (Platform.OS === 'android') {
+          ToastAndroid.show(
+            ACCESSIBILITY_STRINGS.failedToFetchDeletedQuestionaire,
+            ToastAndroid.LONG
+          );
+        } else {
+          Alert.alert(ACCESSIBILITY_STRINGS.failedToFetchDeletedQuestionaire);
+        }
+        let result = false;
+        await ParticipantCode.removeQuestionaireFromLocalStorage(code).then((res) => {
```

Comment:
Hij update pas als ik de app afsluit en opnieuw opstart. Als het kan graag direct een refresh van het scherm. Mocht de vragenlijst niet in local storage staan niks doen en anders de melding geven "De vragenlijst is verwijderd uit de lijst"

Reply:
Ik heb het zojuist nogmaals getest. Als ik eerst de vragenlijst ophaal (dus hij komt ook in localstorage), en hem daarna verwijder en dan weer probeer op te halen dmv de code, dan verdwijnt ie automatisch van het hoofdscherm. 

Daarnaast doet hij ook niets als de vragenlijst niet in localstorage staat (al moet ie daarvoor wel even kijken in local storage en dat doet ie ook in die remove functie).